### PR TITLE
Adding files for homewrok 3

### PR DESCRIPTION
**Pull request by khushikdu**
Adding files for Homework 3.

### CODE REVIEW

**Comment by khushikdu** [comment]
| Metric | Value |
| ------ | ----- |
| alert_status | <b>:white_check_mark: OK</b> |
| bugs | 0 |
| code_smells | 25 |
| reliability_rating | A (0 Bugs)<br/>Note: A being best, E being worst |
| security_rating | A (0 Vulnerabilities)<br/>Note: A being best, E being worst |
| vulnerabilities | 0 |

File: backend/homework/3/pom.xml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    <pluginManagement>
+        <plugins>
+            <plugin>
+                <groupId>org.sonarsource.scanner.maven</groupId>
+                <artifactId>sonar-maven-plugin</artifactId>
+                <version>3.10.0.2594</version>
+            </plugin>
+        </plugins>
+    </pluginManagement>
+</build>
```
Comment (Sagun15) [inline backend/homework/3/pom.xml:40]: Indentation is misaligned

File: backend/homework/3/src/main/java/ques4/BlueCrossBlueShield.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+     * @param plan
+     * @param age
+     * @param smokes
+     * @return
+     */
+    public double computeMonthlyPremium(HealthInsurancePlan plan,int age, boolean smokes){
+
+        double premiuim=0;
+        if(plan instanceof PlatinumPlan){
+            if(age>55)
```
Comment (Sagun15) [inline backend/homework/3/src/main/java/ques4/BlueCrossBlueShield.java:15]: Avoid using nested conditional statements, Prefer defining a separate functions.

File: backend/homework/3/src/main/java/ques4/BronzePlan.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+public abstract class  BronzePlan extends HealthInsurancePlan {
+    public BronzePlan() {
+
+        setCoverage(0.7);
+    }
+
+    @Override
+    public double computeMonthlyPremium(double salary,int age, boolean smokes) {
+        return 0.05*salary+getOfferedBy().computeMonthlyPremium(this, age, smokes);
```
Comment (Sagun15) [inline backend/homework/3/src/main/java/ques4/BronzePlan.java:11]: Give space between different values
return 0.05 * salary + getOfferedBy().computeMonthlyPremium(this, age, smokes);

File: backend/homework/3/src/main/java/ques4/GoldenPlan.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package ques4;
+
+public abstract class GoldenPlan extends HealthInsurancePlan {
+    public GoldenPlan() {
+
```
Comment (Sagun15) [inline backend/homework/3/src/main/java/ques4/GoldenPlan.java:5]: Unnecessary empty line

File: backend/homework/3/src/main/java/ques4/Main.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    public static void main(String[] args) {
+        User staff = new User();
+
+        InsuranceBrand insuranceBrand = new BlueCrossBlueShield();
+        HealthInsurancePlan insurancePlan = new PlatinumPlan();
+
+        insurancePlan.setOfferedBy(insuranceBrand);
+        staff.setInsurancePlan(insurancePlan);
+
+        //insurancePlan.computeMonthlyPremium(15000, 56, true);
```
Comment (Sagun15) [inline backend/homework/3/src/main/java/ques4/Main.java:18]: Remove this comment

**Review by Sagun15 — COMMENTED** [review]
Reach out to me if you have any doubts or concerns

**Pull request merged**